cli/pcluster/cli_commands/start.py

```diff
@@ -0,0 +1,129 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+# with the License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+import abc
+import logging
+import sys
+from abc import abstractmethod
+
+import boto3
+from botocore.exceptions import ClientError
+
+from pcluster import utils
+from pcluster.cli_commands.compute_fleet_status_manager import ComputeFleetStatus, ComputeFleetStatusManager
+from pcluster.config.pcluster_config import PclusterConfig
+from pcluster.utils import error
+
+if sys.version_info >= (3, 4):
+    ABC = abc.ABC
+else:
+    ABC = abc.ABCMeta("ABC", (), {})
+
+LOGGER = logging.getLogger(__name__)
+
+
+def start(args):
+    """Start cluster compute fleet."""
+    pcluster_config = PclusterConfig(config_file=args.config_file, cluster_name=args.cluster_name, auto_refresh=False)
+    cluster_section = pcluster_config.get_section("cluster")
+    scheduler = cluster_section.get_param_value("scheduler")
+
+    SCHEDULER_TO_START_COMMAND_MAP[scheduler]().start(args, pcluster_config)
+
+
+class StartCommand(ABC):
```

Comment:
I like this idea.

It seems to me that in future this logic should be moved to the ClusterModel and have the start/stop command implementation that depends on the cluster model (awsbatch should became another Cluster Model).

I'm saying so because the initialization part is very similar to the one in the stop.py so we could put this logic in common.

Reply:
I agree on finding a way to not duplicate part of the logic. What I don't like about moving all commands to ClusterModel is that we are going to have a single class implementing all commands while I find it useful to have separate contained modules and classes for each command.